packages/api/src/client/utils/httpClient.ts

```diff
@@ -60,12 +61,16 @@ export class HttpClient implements IHttpClient {
     this.baseUrl = opts.baseUrl;
     // A higher default timeout, validator will sets its own shorter timeoutMs
     this.timeoutMs = opts.timeoutMs ?? 60_000;
-    this.getAbortSignal = opts.getAbortSignal;
+    this.signal = opts.signal;
     this.fetch = opts.fetch ?? fetch;
     this.metrics = metrics ?? null;
     this.logger = logger ?? null;
   }
 
+  setAbortSignal(signal: AbortSignal): void {
```

Comment:
should this only be allowed once? Eg: `if (this.signal) throw .. else this.signal = signal`

Reply:
No, to allow the validator client to work it must allowed N times. Every time the validator is started a new signal is created, so it must be set to the HttpClient every time.

Reply:
I forget that our validator can start and stop (and restart). I still think its weird.